### PR TITLE
Feat/#2200 allow modifying stake fraction

### PR DESCRIPTION
**Pull request by davecreaser**
## Description

- Create saga to allow modification of stake fraction.
- UI to follow in another PR.

Block ingestor PR here: https://github.com/JoinColony/block-ingestor/pull/223

Ticket here: https://github.com/JoinColony/colonyCDapp/issues/2200

## Testing

Create a fake UI to trigger the saga, and trigger it. The stake fraction for the staked expenditure extension should be updated in the db.

## Diffs

**New stuff** ✨

* New saga 'setStakeFraction'

Contributes to #2200

### CODE REVIEW

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice work on this Dave! All working as expected 👍 

Screenshot 2024-05-17 at 16 12 33

Just looks like a couple of unrelated commits got accidentally mixed into this, once those are resolved should be good to go!

**Comment by iamsamgibbs** [comment]
Some code snippets to add to `TmpAdvancedPayments.tsx` for the sake of anyone else who wants to test:

`import { type SetStakeFractionPayload } from '~redux/sagas/expenditures/setStakeFraction.ts';`

```
  const setStakeFraction = useAsyncFunction({
    submit: ActionTypes.SET_STAKE_FRACTION,
    error: ActionTypes.SET_STAKE_FRACTION_ERROR,
    success: ActionTypes.SET_STAKE_FRACTION_SUCCESS,
  });

  const handleModifyStakeFraction = async () => {
    const payload: SetStakeFractionPayload = {
      colonyAddress: colony.colonyAddress,
      stakeFraction: '50',
    };

    await setStakeFraction(payload);
  };
  ```
  
  ` <Button onClick={handleModifyStakeFraction}>Modify stake fraction</Button>`

**Comment by davecreaser** [comment]
> Nice work on this Dave! All working as expected 👍
> 
> Screenshot 2024-05-17 at 16 12 33
> Just looks like a couple of unrelated commits got accidentally mixed into this, once those are resolved should be good to go!

Cheers Sam! And thanks for catching those commits, I've rebased and removed them :)

**Review by iamsamgibbs — APPROVED** [review]

**Review by rdig — APPROVED** [review]
Nicely done!

File: src/i18n/en.json [hunk 1]
```diff
@@ -422,6 +424,10 @@
     "transaction.group.addVerifiedMembers.description": "Add verified members",
     "transaction.group.removeVerifiedMembers.title": "Remove verified members",
     "transaction.group.removeVerifiedMembers.description": "Remove verified members",
+    "transaction.group.setStakeFraction.title": "Set Stake Fraction",
+    "transaction.group.setStakeFraction.description": "Set Stake Fraction",
+    "transaction.ColonyClient.installExtension.title": "Install extension",
+    "transaction.ColonyClient.installExtension.description": "Install extension",
```
Comment (jakubcolony) [inline src/i18n/en.json:430]: Nice catch!

**Review by jakubcolony — APPROVED** [review]
Changes look and work great, nice job @davecreaser 👍 

Considering it's resolving the first part of the issue, would you mind opening another one for the other subtask?

**Comment by davecreaser** [comment]
> Changes look and work great, nice job @davecreaser 👍
> 
> Considering it's resolving the first part of the issue, would you mind opening another one for the other subtask?

@jakubcolony No problem. You mean opening another issue or another PR?

**Pull request merged**